04_numeronym/Jat/numeroniyum_test.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestNumeroniumMainOutput(t *testing.T) {
+	// Given
+	r := require.New(t)
+	var buf bytes.Buffer
+	out = &buf
+
+	// When
+	main()
+
+	// Then
+	expected := "[ ab abc a3a K8s]\n"
+
+	actual := buf.String()
+	r.Equalf(expected, actual, "Unexpected output in main()")
+}
+func TestNumeroniumsNoArgs(t *testing.T) {
+	// Given
+	r := require.New(t)
+
+	// When
+	m := numeronyms()
+
+	// Then
+	expected := 0
+	actual := len(m)
+	r.Equalf(expected, actual, "Unexpected output in main()")
```

Comment:
Bad failure message. 
Try setting up using [table driven tests](https://github.com/golang/go/wiki/TableDrivenTests) and have as one of the "row" values the failure message - that should simplify/shorten your code a lot.

Also make sure your code works with unicode, e.g. emojis

Reply:
Add test cases for unicode at beginning and end and white space at beginning and end please